domain/consensus/processes/dagtraversalmanager/dagtraversalmanager_test.go

```diff
@@ -3,11 +3,137 @@ package dagtraversalmanager_test
 import (
 	"github.com/kaspanet/kaspad/domain/consensus"
 	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/consensus/model/testapi"
 	"github.com/kaspanet/kaspad/domain/consensus/utils/testutils"
 	"github.com/kaspanet/kaspad/domain/dagconfig"
 	"testing"
 )
 
+const commonChainSize = 5
+const depth uint64 = 2
+
+func TestBlockAtDepthOnChainDag(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+		factory := consensus.NewFactory()
+		tc, tearDown, err := factory.NewTestConsensus(params, false,
+			"TestBlockAtDepthOnChainDag")
+		if err != nil {
+			t.Fatalf("Failed creating a NewTestConsensus: %s", err)
+		}
+		defer tearDown(false)
+
+		// This test compares the result of BlockAtDepth to the result of looping over the SelectedChain on a single chain DAG.
```

Comment:
This comment should be on top of the function declaration, in the format of `// TestBlockAtDepthOnChainDag compares the result...`

(The same for all other comments that describe what a test does)